api/src/main/java/org/pmiops/workbench/leonardo/LeonardoApiClientImpl.java

```diff
@@ -439,14 +439,14 @@ public void updatePersistentDisk(String googleProject, String diskName, Integer
   @Override
   public List<LeonardoListPersistentDiskResponse> listPersistentDiskByProjectCreatedByCreator(
```

Comment:
I think we could go two ways with this.  If we do combine these, then I think we should also remove the indirection of calling the private `listPersistentDiskByProject()` and handle it all in this method.

OTOH: I like this separation.  We have a binary choice between "the normal flow" and "admin-flow" with a fixed set of parameters for each.  Even `includeDeleted` is a binary choice between admin and user-flow: it's always false for the latter, if you trace how it's called.  I want to remove the private `listPersistentDiskByProject()` in this case too: all of its logic is directed to making that same binary choice.

Reply:
Hmmm, I agree with the second approach, but I have some questions about the second approach. 

I went with option two for now, but instead of getting rid of `listPersistentDiskByProject()`I changed its parameters to accept a `Provider<DisksApi>` and a String for a Leonardo. In that case, there is no admin logic in the function, it is just the logic that the two flows in common. In an attempt not to duplicate code, this common code is in a shared function. What do you think about that?